plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammar_fstrings/FStringsGrammarFactory.java

```diff
@@ -0,0 +1,25 @@
+package org.python.pydev.parser.grammar_fstrings;
+
+import org.python.pydev.parser.jython.FastCharStream;
+
+public class FStringsGrammarFactory {
+
+    public static FStringsGrammar createGrammar(String s) {
```

Comment:
You could have a different method and test just the string replacement directly (so, create a `public static char[] replaceNamedUnicodeVars(String s)` which will return a `char[]` with the return and test that directly to see input/output (instead of just testing checking the vars as it's now).

And this method would call that one when creating the grammar ;)